### ISSUE TITLE
Keybindings for snippets

### ISSUE THREAD

**Issue by paularmstrong** [zed-industries/zed#20036]
### Check for existing issues

- [X] Completed

### Describe the feature

**Context:**

In vscode, sublime, textmate, bbedit… for 20+ years I've had a snippet configured with key bindings that looks like this (vscode):

```
{
  "html tag": {
    "scope": "javascriptreact,typescriptreact,html,astro",
    "body": "<${1:div}${2}>${3:$TM_SELECTED_TEXT}</$1>",
    "description": "Create an HTML/JSX tag"
  },
}
```

In vscode keybindings:

```
[
  {
    "key": "ctrl+cmd+,",
    "command": "editor.action.insertSnippet",
    "when": "editorTextFocus",
    "args": {
      "langId": "javascriptreact",
      "name": "html tag"
    }
  }
]
```

**Feature**

The ability to trigger a snippet with tabbable placeholder macros via key bindings, just like other editors.

**Comment by notpeter** [comment]
Have you looked [Zed Snippets documentation](https://zed.dev/docs/snippets) or the Emmet Extension?  Both allow you to define snippets where the language server will replace typed shortcuts with snippet blocks.

**Comment by paularmstrong** [comment]
> Have you looked [Zed Snippets documentation](https://zed.dev/docs/snippets) or the Emmet Extension? Both allow you to define snippets where the language server will replace typed shortcuts with snippet blocks.

Yes, and that doesn't have any information about possibly using keybindings.

**Comment by snnsnn** [comment]
@JosephTLyons @notpeter Hi, could you please prioritize this feature?
It should be simple to implement — just a command binding — and would significantly boost productivity. This is one of the main blockers for migrating from VS Code. Thanks!

Also, the issue is currently marked as "awaiting info," but the original post already includes a clear explanation and rationale. It seems ready for consideration as-is.

**Comment by emilycares** [comment]
Yes this shortcut does not exist in 0.212.7

**Comment by snnsnn** [comment]
@zelenenka Hi! I’m not sure how much vote counts affect prioritization (for example: [https://github.com/zed-industries/zed/issues/5393](https://github.com/zed-industries/zed/issues/5393)), but I wanted to underline how important this feature is for anyone who relies on keyboard-triggered snippets. It’s a core part of my workflow, and not having it really slows down my typing and overall flow. I’d love to switch to Zed full-time, but this one limitation makes that difficult.

This isn’t just a user-facing convenience either—it’s a fundamental capability for plugins. Snippets with context-aware behavior can adapt to where you are in the code and provide a huge productivity boost. That level of flexibility is essential for advanced tooling and editor extensions.

I know Zed has many feature requests to juggle, but this feels like something foundational that should take priority over more cosmetic additions like rainbow brackets. I’d really appreciate if this could get another look.
